app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +138,141 @@ object DialogUtils {
             }
         }
     }
+
+    fun colorPickerBottomSheet(
+        context: Context,
+        previousColor: Hsv?,
+        /**
+         * Lambda that provides the string of the current color.
+         * Returns nothing. Used to confirm the use of the color.
+         */
+        onSubmit: ((Hsv?) -> Unit)
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val view = LayoutInflater.from(context).inflate(R.layout.color_dialog, null)
+        val huePicker = view.findViewById<HueView>(R.id.huePicker)
+        val gradientPicker = view.findViewById<GradientView>(R.id.gradientPicker)
+        val hexText = view.findViewById<ForcableEditText>(R.id.hexText)
+        val hueText = view.findViewById<ForcableEditText>(R.id.hueText)
+        val satText = view.findViewById<ForcableEditText>(R.id.satText)
+        val valText = view.findViewById<ForcableEditText>(R.id.valText)
+        val redText = view.findViewById<ForcableEditText>(R.id.redText)
+        val greenText = view.findViewById<ForcableEditText>(R.id.greenText)
+        val blueText = view.findViewById<ForcableEditText>(R.id.blueText)
+        val topColorSquare = view.findViewById<View>(R.id.topColor)
+        val bottomColorSquare = view.findViewById<View>(R.id.bottomColor)
+        val confirmButton = view.findViewById<Button>(R.id.okButton)
+        val allText = listOf(hexText, hueText, satText, valText, redText, greenText, blueText)
+        confirmButton.setOnClickListener {
+            onSubmit.invoke(
+                Hsv(
+                    hueText.floatValue,
+                    satText.floatValue / 100,
+                    valText.floatValue / 100
+                )
+            )
+            sheet.dismiss()
+        }
+        val cancelButton = view.findViewById<Button>(R.id.cancelButton)
+        cancelButton.setOnClickListener {
+            onSubmit.invoke(null)
+            sheet.dismiss()
+        }
+        sheet.setOnCancelListener {
+            onSubmit.invoke(null)
+            sheet.dismiss()
+        }
+        /**
+         * Lambda that updates all the views in the gradient.
+         * Takes in a color state data class to update views.
+         */
+        val forceViewUpdates: ((Hsv) -> Unit) = { hsv ->
+            val rgb = calculateColorWithHsv(hsv)
+            val hex = calculateHex(rgb)
+            val saturation = hsv.saturation * 100
```

Comment:
Why not just use `hsv.saturation` inline instead of making a variable? Same for `value`? You even do this on line 203

Reply:
Changed